specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,1028 @@
+merged and how, additional module flags may be added to the bitcode only if
+their behavior is set to `Warning`, `Append`, `AppendUnique`, `Max`, or `Min`.
+It is at the discretion of the maintainers for various components in the QIR
+stack to discard module flags that are not explicitly required or listed as
+optional flags in the QIR specification.
+
+### Specification Version
+
+The required flags `"qir_major_version"` and `"qir_minor_version"` identify the
+major and minor versions of the specification that the QIR bitcode adheres to.
+
+- Since the QIR specification may introduce breaking changes when updating to a
+  new major version, the behavior of the `"qir_major_version"` flag must be set
+  to `Error`; merging two modules that adhere to different major versions must
+  fail.
+
+- The QIR specification is intended to be backward compatible within the same
+  major version but may introduce additional features as part of newer minor
+  versions. The behavior of the `"qir_minor_version"` flag must hence be `Max`
+  so that merging two modules compiled for different minor versions results in a
+  module that adheres to the newer of the two versions.
+
+### Memory Management
+
+Each bitcode file contains the information whether pointers of type `%Qubit*`
+and `%Result*` point to a valid memory location, or whether a pointer merely
+encodes an integer constant that identifies which qubit or result the value
+refers to. This information is represented in the form of the two module flags
+named `"dynamic_qubit_management"` and `"dynamic_result_management"`. Within the
+same bitcode module, there can never be a mixture of the two different kinds of
+pointers. The behavior of both module flags correspondingly must be set to
+`Error`. As detailed in the section on [qubit and result
+usage](#qubit-and-result-usage), an adaptive profile-compliant program must not make
+use of dynamic qubit or result management. The value of both module flags hence
+must be set to `false`.
+
+For `i1`, `i64`, `f64`, ... values created by mid-circuit measurement, extern
+functions, or classical computations the assumption is that while a `%Result*`
+may point to a valid memory location in RAM or some other memory pool, by default,
+instructions performed on virtual registers with these data types correspond to
+these values being stored in integer or floating registers when an instruction is
+executed. Before a virtual register is used in an instruction, there is no
+assumption that the value in the virtual register always corresponds to a physical
+register. For example, when considering register coloring, the virtual register,
+`%0`, in the QIR program may refer to a value stored in RAM for most of its
+lifetime before being loaded into a register when an instruction operates on `%0`.
+backends should specify any constraints on classical compute support on this
+[page](./Adaptive_Hardware/providers.md#backend-support-for-adaptive-profile).
+
+### Error Messages
```

Comment:
I think this section would fit better into a subsection under ## Entry Point Definition
```suggestion
### Error Codes
```